### PR TITLE
AdminUI: Select Improvements

### PR DESCRIPTION
**Pull request by steve-keep**
https://github.com/exogee-technology/graphweaver/assets/85143753/52e8bba4-36fd-4a7f-8a91-2f110267bb07

### CODE REVIEW

**Review by thekevinbrown — APPROVED** [review]
Nothing worth blocking over, looking great!

File: src/packages/admin-ui-components/src/select/component.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+				return _selectedOptions;
+			});
+		}
+	}
+
+	// This is only used when multi select is enabled
+	const handleDeleteAll = () => setSelectedItems([]);
+
+	const handleOnPillKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+		if (mode !== SelectMode.MULTI) return;
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/admin-ui-components/src/select/component.tsx:78]: sounds good, I will also update to allow the `x` inside the pill so you can clear a single select 👍

File: src/packages/admin-ui-components/src/select/component.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	placeholder = 'Select',
+	loading = false,
+	mode,
+}: SelectProps) => {
+	const [selectedItems, setSelectedItems] = useState<SelectOption[]>(value);
+
+	const { isOpen, getToggleButtonProps, getMenuProps, highlightedIndex, getItemProps } = useSelect({
+		items: options,
+		onSelectedItemChange: handleSelectionChange,
+		itemToString: (item) => (item?.label ? item.label : ''),
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/admin-ui-components/src/select/component.tsx:42]: old habits die hard 😄

File: src/packages/admin-ui-components/src/select/component.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+			{isOpen && (
+				<ul className={styles.optionsDropdown} {...getMenuProps()}>
+					{loading ? (
+						<Spinner />
+					) : (
+						options.map((item, index) => (
+							<li
+								className={`${highlightedIndex === index ? styles.highlighted : ''} ${
+									selectedIds.includes(item.value) ? styles.selected : ''
+								} ${styles.option}`}
```
[the remark this answers is not in the document]
Comment (steve-keep) [reply src/packages/admin-ui-components/src/select/component.tsx:126]: yeah fair the formatting makes it hard to read.

**Pull request merged**